compiler/modules/nback/structure.bal

```diff
@@ -314,14 +345,27 @@ function buildSpecializedListSet(llvm:Builder builder, Scaffold scaffold, llvm:V
                    bbTaggedSet);
     builder.positionAtEnd(bbSpecializedSet);
     llvm:PointerValue array = <llvm:PointerValue>builder.load(builder.getElementPtr(struct, [llvm:constInt(LLVM_INT, 0), llvm:constInt(LLVM_INDEX, 3)]), ALIGN_HEAP);
-    array = builder.bitCast(array, heapPointerType(llvm:arrayType(memberTy, 0)));
-
-    builder.store(val, builder.getElementPtr(array, [llvm:constInt(LLVM_INT, 0), index], "inbounds"));
+    array = builder.bitCast(array, heapPointerType(llvm:arrayType(repr.memberHeapType, 0)));
+    llvm:Value heapVal = convertValue(builder, val, repr.memberType, repr.memberHeapType);
+    builder.store(heapVal, builder.getElementPtr(array, [llvm:constInt(LLVM_INT, 0), index], "inbounds"));
     builder.br(bbJoin);
     builder.positionAtEnd(bbTaggedSet);
     return bbJoin;
 }
 
+function convertValue(llvm:Builder builder, llvm:Value value, llvm:Type fromTy, llvm:Type toTy) returns llvm:Value {
```

Comment:
I think it's better to have two functions listReprConvertTo/FromHeapType(ListRepr, Builder, Value)